src/Classes/Particle/Updater.ts

```diff
@@ -214,57 +262,45 @@ export class Updater {
-                particle.velocity.horizontal = -particle.velocity.horizontal + (particle.velocity.vertical / 2);
-                particle.velocity.vertical = -particle.velocity.vertical + (particle.velocity.horizontal / 2);
+                this.polygonBounce();
             }
         } else if (options.polygon.type === PolygonMaskType.inline) {
             if (!particle.initialPosition) {
                 particle.initialPosition = {x: 0, y: 0};
             }
-
-            const dx = particle.initialPosition.x - particle.position.x;
-            const dy = particle.initialPosition.y - particle.position.y;
-            const dist = Math.sqrt(dx * dx + dy * dy);
-
-            if (dist > options.polygon.move.radius) {
-                particle.velocity.horizontal = -particle.velocity.horizontal + (particle.velocity.vertical / 2);
-                particle.velocity.vertical = -particle.velocity.vertical + (particle.velocity.horizontal / 2);
+            const dist = Utils.getDistanceBetweenCoordinates(particle.initialPosition, particle.position);
+            if (dist > container.retina.polygonMaskMoveRadius) {
+                this.polygonBounce();
             }
         } else {
+            const outMode = options.particles.move.outMode;
             const x = particle.position.x + particle.offset.x;
             const y = particle.position.y + particle.offset.y;
 
-            if (x + particle.radius > container.canvas.dimension.width || x - particle.radius < 0) {
-                particle.velocity.horizontal = -particle.velocity.horizontal;
+            if (outMode === OutMode.bounce || outMode === OutMode.bounceHorizontal) {
+                Updater.checkBounds(x, particle.radius, container.canvas.dimension.width, () => {
+                    particle.velocity.horizontal = -particle.velocity.horizontal;
+                });
             }
 
-            if (y + particle.radius > container.canvas.dimension.height || y - particle.radius < 0) {
-                particle.velocity.vertical = -particle.velocity.vertical;
+            if (outMode === OutMode.bounce || outMode === OutMode.bounceVertical) {
+                Updater.checkBounds(y, particle.radius, container.canvas.dimension.height, () => {
+                    particle.velocity.vertical = -particle.velocity.vertical;
+                });
             }
         }
     }
 
-    private updateBounceVertical(): void {
-        const container = this.container;
-        const particle = this.particle;
-
-        if (particle.position.y + particle.radius > container.canvas.dimension.height) {
-            particle.velocity.vertical = -particle.velocity.vertical;
-        }
-
-        if (particle.position.y - particle.radius < 0) {
-            particle.velocity.vertical = -particle.velocity.vertical;
+    private static checkBounds(coordinate: number, radius: number, size: number, outside: () => void): void {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Declaration of private static method not allowed after declaration of private instance method. Instead, this should come after constructors.](https://app.codacy.com/manual/ar3s/tsparticles/pullRequest?prid=5139404)